### ISSUE TITLE
kiali helm chart ingress apiVersion check error

### ISSUE THREAD

**Issue by TIEDPAG** [kiali/kiali#4267]
**General remarks**
我尝试在k8s 1.18.17上使用helm安装kiali，出现了一个错误

**Describe the bug**
错误如下：
> Error: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "Ingress" in version "networking.k8s.io/v1"

查看了chart/template/Ingress.yaml文件，发现
```yaml
{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1" }}
apiVersion: networking.k8s.io/v1
{{- else }}
apiVersion: networking.k8s.io/v1beta1
{{- end }}
```
这里似乎helm内置变量使用错误导致的，应该是
```yaml
{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1.Ingress" }}
apiVersion: networking.k8s.io/v1
{{- else }}
apiVersion: networking.k8s.io/v1beta1
{{- end }}
```

**Versions used**
Kiali: latest
Istio: 1.10.3
Kubernetes flavour and version: kubeadm install 1.18.17

**To Reproduce**
1. add kiali repo to helm
2. helm install kiali kiali/kiali-server

**Expected behavior**
kiali安装完成，并可以使用nginx-ingress访问

**Comment by jmazzitelli** [comment]
> ```yaml
> {{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1.Ingress" }}
> ```

I do not know where you see this. Note that this was [recently changed](https://github.com/kiali/helm-charts/pull/95/files
) but it is "v1/Ingress" not "v1.Ingress":

There is [no where in the helm charts where "v1.Ingress"](https://github.com/kiali/helm-charts/search?q=v1.Ingress) is found.

1. What is the EXACT version of the kiali server helm chart you are using?
2. What is the EXACT command line you run to execute/install the helm chart?

**Comment by TIEDPAG** [comment]
是的，应该是"v1/Ingress"

**Comment by jmazzitelli** [comment]
I cannot read the non-English characters. But if I understand the problem, I can tell you the latest Kiali helm chart should have this fixed as per https://github.com/kiali/helm-charts/pull/95/files

Please try the latest helm chart that was released yesterday, v1.38.1.

**Comment by jmazzitelli** [comment]
cc @nelson-lark can you look at this? This should have been fixed by your recent PR, no?

**Comment by nelson-lark** [comment]
@jmazzitelli Yes, this was the precise issue I had, and on the same version of Kubernetes.  From translate.google.com, they are suggesting the same fix.

Master branch has the fix merged, but the latest tag v1.38.1 looks to be missing it.
https://github.com/kiali/helm-charts/blob/master/kiali-server/templates/ingress.yaml
https://github.com/kiali/helm-charts/blob/v1.38.1/kiali-server/templates/ingress.yaml

**Comment by jmazzitelli** [comment]
> Master branch has the fix merged, but the latest tag v1.38.1 looks to be missing it.

Oh, right! Yes, only a few things were cherry-picked to the v1.38 branch to make it into the .1 patch release. This one did not make the cut.

But, yes, this fix will be in the next release then - v1.39 - due to be released in 2 weeks (August 20, 2021).

I'll close this as a duplicate issue and is considered fixed via https://github.com/kiali/helm-charts/pull/95
